pytorch_lightning/trainer/trainer.py

```diff
@@ -2612,6 +2613,36 @@ def weights_summary(self, val: Optional[str]) -> None:
         rank_zero_deprecation("Setting `Trainer.weights_summary` is deprecated in v1.5 and will be removed in v1.7.")
         self._weights_summary = val
 
+    """
+    Estimator Properties
+    """
+
+    @property
+    def estimated_num_training_steps(self) -> int:
+        """Total training steps inferred from dataloaders and distributed setup."""
+        # infinite training
+        if self.max_epochs == -1 and self.max_steps == -1:
+            return float("inf")
+
+        if self.train_dataloader is None:
+            rank_zero_warn("Loading `train_dataloader` to estimate number of training steps.")
+            self.reset_train_dataloader()
+
+        total_batches = self.num_training_batches
+
+        # iterable dataset
+        if total_batches == float("inf"):
+            return self.max_steps
+
+        self.accumulate_grad_batches = self.accumulation_scheduler.get_accumulate_grad_batches(self.current_epoch)
```

Comment:
I'm worried about this. If someone uses an actual scheduler, the value of `estimated_num_training_steps` will change every epoch. Is that meaningful? Since it's a property and not a method, it may not be obvious that this value gets recomputed every epoch (if I access it every epoch). 

Three options 
1) document that behavior 
2) since the scheduling is finite, one could give a better estimate by integrating the scheduling into the computation here (so we essentially wouldn't depend on current_epoch)
3) Only allow accumulation with a constant

Reply:
I'd say 3rd one? since the only use-case I currently know of is that this is being used within a scheduler and the accumulation factor is assumed to be constant in such a case.

Reply:
Yes, that's acceptable imo.